crates/hyperswitch_connectors/src/connectors/novalnet/transformers.rs

```diff
@@ -125,99 +321,687 @@ impl<F, T> TryFrom<ResponseRouterData<F, NovalnetPaymentsResponse, T, PaymentsRe
     fn try_from(
         item: ResponseRouterData<F, NovalnetPaymentsResponse, T, PaymentsResponseData>,
     ) -> Result<Self, Self::Error> {
+        match item.response.result.status {
+            NovalnetAPIStatus::SUCCESS => {
+                let redirection_data: Option<RedirectForm> =
+                    item.response
+                        .result
+                        .redirect_url
+                        .map(|url| RedirectForm::Form {
+                            endpoint: url.to_string(),
+                            method: Method::Get,
+                            form_fields: HashMap::new(),
+                        });
+
+                let transaction_id = item
+                    .response
+                    .transaction
+                    .clone()
+                    .and_then(|data| data.tid.map(|tid| tid.to_string()));
+                let transaction_status = item
+                    .response
+                    .transaction
+                    .and_then(|transaction_data| transaction_data.status)
+                    .unwrap_or(NovalnetTransactionStatus::PROGRESS);
```

Comment:
By default it should be `AuthenticationPending`? Does not seems right

Reply:
The` transaction.status` for Novalnet is optional . The response has two parts `result` and `transaction`.

 If the `result.status` is `SUCCESS`, we should always get a `result.redirect_url` in 3ds flow but `transaction.status` is optional. So that is why I mapped it with default `AuthenticationPending`. After redirection is successful, it should get status `Authorized`